### PR TITLE
Sources data model

### PR DESCRIPTION
**Pull request by vabd**

### CODE REVIEW

File: content/information-distribution/source.md [hunk 1]
```diff
@@ -4,13 +4,48 @@ weight: 2
...
 A source is an entity holding the responsibility of publishing information
-through the Matrix room. Before publishing any information, a source **must**
-register itself as such on the room, including data on the organism or
-individual operating the source, the publication language and the publication
-name space (2️⃣: a Matrix event type), along with one or more public signature
-verification keys that will be used to assert the integrity of the information
-it publishes, and the signing algorithm used to generate the signature of its
-articles (`ed25519`, `hmac-sha256`, `hmac-sha512`, etc.).
+through the Matrix room. Each source **must** be a registered Matrix user. A
+Matrix user **must not** register more than one source.
```
Comment (GordonF42) [inline content/information-distribution/source.md]: It's more a limitation rather than a rule. One source <=> one user, one user <=> one source.

File: content/information-distribution/source.md [hunk 2]
```diff
@@ -4,13 +4,48 @@ weight: 2
...
+
+Before publishing any information, a source **must** register itself as such on
+the room. This **must** be done through the publication of a state event of the
```
Comment (GordonF42) [inline content/information-distribution/source.md]: This is already covered differently in https://github.com/Informo/specs/blob/master/content/information-distribution/article.md#matrix-event-networkinformoarticle
> If the sender is not a source user, the article should be ignored. If the source registers itself afterwards, its previously sent articles should become visible.

File: content/information-distribution/source.md [hunk 3]
```diff
@@ -4,13 +4,48 @@ weight: 2
...
+`network.informo.source` type. The event's state key **must** be the ID of the
+source's Matrix user. The content of this event **must** be provided using the
+following model:
+
+|      Parameter      |    Type    | Req. |                                                        Description                                                       |
+| ------------------- | ---------- | :--: | ------------------------------------------------------------------------------------------------------------------------ |
+| `name`              | `string`   |  x   | Name of the source.                                                                                                      |
+| `origin`            | `string`   |  x   | The company or individual maintaining this source.                                                                       |
```
Comment (GordonF42) [inline content/information-distribution/source.md]: `origin` is a bit ambiguous IMO. Maybe `owner`, `administrator`, `maintainer`, ...

File: content/information-distribution/source.md [hunk 4]
```diff
@@ -4,13 +4,48 @@ weight: 2
...
+| `langs`             | `[lang]`   |  x   | Languages of the source's publications.                                                                                  |
+| `sig_algo`          | `string`   |  x   | Algorithm the source will use to cryptographically sign its articles. 🔧                                                  |
+| `sig_keys`          | `[string]` |  x   | Public keys the source will use to cryptographically sign its articles. 🔧                                                |
+| `website`           | `string`   |      | URL of the source's website, if there's one.                                                                             |
+| `description`       | `string`   |      | Short description of the source and its publications.                                                                    |
+| `logo`              | `string`   |      | Logo of the source. If provided, must be a [`mxc://` URL](https://matrix.org/docs/spec/client_server/r0.4.0.html#id112). |
```
Comment (GordonF42) [inline content/information-distribution/source.md]: maybe fallback to the matrix user's avatar?

File: content/information-distribution/source.md [hunk 5]
```diff
@@ -4,13 +4,48 @@ weight: 2
...
+| `name`              | `string`   |  x   | Name of the source.                                                                                                      |
+| `origin`            | `string`   |  x   | The company or individual maintaining this source.                                                                       |
+| `langs`             | `[lang]`   |  x   | Languages of the source's publications.                                                                                  |
+| `sig_algo`          | `string`   |  x   | Algorithm the source will use to cryptographically sign its articles. 🔧                                                  |
+| `sig_keys`          | `[string]` |  x   | Public keys the source will use to cryptographically sign its articles. 🔧                                                |
+| `website`           | `string`   |      | URL of the source's website, if there's one.                                                                             |
+| `description`       | `string`   |      | Short description of the source and its publications.                                                                    |
```
Comment (GordonF42) [inline content/information-distribution/source.md]: Maybe we need to store description as a localized string, in order to allow a subsource provide specific information if it has different management, content policies, etc. than its other peer sources

**Comment by vabd** [comment]
Some elements I missed to include in this SCS:

* The location of sub-sources in trust networks.
* A section on client implementations behaviours.

I'll add these later on.

It might also be interesting to notice that the events properties that are related to signatures might be inconsistent with the naming in other parts of the specs. Basically, we're waiting on #7 (i.e. figuring out whether relying on Matrix's cryptographic features to sign our events is possible) to be complete before harmonising this whole part of the specs and adding some technical details and advices on signature generation.

File: content/information-distribution/source.md [hunk 6]
```diff
@@ -4,13 +4,48 @@ weight: 2
...
+
+Before publishing any information, a source **must** register itself as such on
+the room. This **must** be done through the publication of a state event of the
+`network.informo.source` type. The event's state key **must** be the ID of the
+source's Matrix user. The content of this event **must** be provided using the
+following model:
+
+|      Parameter      |    Type    | Req. |                                                        Description                                                       |
+| ------------------- | ---------- | :--: | ------------------------------------------------------------------------------------------------------------------------ |
+| `name`              | `string`   |  x   | Name of the source.                                                                                                      |
```
Comment (GordonF42) [inline content/information-distribution/source.md]: Source name must be localize-able, so that for example, a russian source (with cyrillic characters) can display an english name (with latin characters)

File: content/information-distribution/source.md [hunk 7]
```diff
@@ -4,13 +4,48 @@ weight: 2
...
+
+ <!-- 🔧: Need to do some research on Megolm and Matrix APIs around encryption
+ and key management -->
+
+Where `lang` is an object using the following model:
+
+|      Parameter      |    Type    | Req. |                                                                       Description                                                                        |
+| ------------------- | ---------- | :--: | -------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `locale`            | `string`   |  x   | Identifier of the language (and variant) in which the articles are written. **Must** be compliant with [RFC 5646](https://tools.ietf.org/html/rfc5646).  |
+| `mxid`              | `string`   |  x   | The Matrix identifier of the sub-source publishing the articles. **Can** be the current source.                                                          |
```
Comment (GordonF42) [inline content/information-distribution/source.md]: What the point of being both a source and a sub-source? can you add an example?
Edit: After reading more I found the reason. Maybe the `**Can** be the current source` needs some precisions  /could be omitted / explained below this table

File: content/information-distribution/source.md [hunk 8]
```diff
@@ -4,13 +4,48 @@ weight: 2
...
+| `langs`             | `[lang]`   |  x   | Languages of the source's publications.                                                                                  |
+| `sig_algo`          | `string`   |  x   | Algorithm the source will use to cryptographically sign its articles. 🔧                                                  |
+| `sig_keys`          | `[string]` |  x   | Public keys the source will use to cryptographically sign its articles. 🔧                                                |
+| `website`           | `string`   |      | URL of the source's website, if there's one.                                                                             |
+| `description`       | `string`   |      | Short description of the source and its publications.                                                                    |
+| `logo`              | `string`   |      | Logo of the source. If provided, must be a [`mxc://` URL](https://matrix.org/docs/spec/client_server/r0.4.0.html#id112). |
+| `custom`            | `object`   |      | Additional information for custom client implementations.                                                                |
```
Comment (GordonF42) [inline content/information-distribution/source.md]: Maybe add something about the source location or country, so users can filter the source list to show only sources related to his country

File: content/information-distribution/source.md [hunk 9]
```diff
@@ -4,13 +4,48 @@ weight: 2
...
+
+The array of `lang` objects within the source's registration **must** contain at
+least one element. More information on localisation is available
+[below](#localisation).
+
+The registration event is the one being signed by [trust
+authorities](/trust-management/trust-authority). Each time one of the source's
+properties changes, a new registration event **must** be published, and a new
+signature **must** be generated by each of the trust authorities certifying the
+trustworthiness of this source.
```
Comment (GordonF42) [inline content/information-distribution/source.md]: rewording:
> Each time one of the source's properties changes, a new registration event **must** be published, and every trust authority certifying the trustworthiness this source **must** update their signatures accordingly.

File: content/information-distribution/source.md [hunk 10]
```diff
@@ -4,13 +4,48 @@ weight: 2
...
+
+## Compromission of cryptographic private keys
 
 A source **must** be trusted by trust authorities operated by organisms it
```
Comment (GordonF42) [inline content/information-distribution/source.md]: This is outside of your PR, but the `**must**` should be replaced with a `**should**`: A new source will not be trusted immediately by TAs (untrusted sources can exist but will be presented to the user as particularly unsafe)

File: content/information-distribution/source.md [hunk 11]
```diff
@@ -4,13 +4,48 @@ weight: 2
...
+|      Parameter      |    Type    | Req. |                                                                       Description                                                                        |
+| ------------------- | ---------- | :--: | -------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `locale`            | `string`   |  x   | Identifier of the language (and variant) in which the articles are written. **Must** be compliant with [RFC 5646](https://tools.ietf.org/html/rfc5646).  |
+| `mxid`              | `string`   |  x   | The Matrix identifier of the sub-source publishing the articles. **Can** be the current source.                                                          |
+
+The array of `lang` objects within the source's registration **must** contain at
+least one element. More information on localisation is available
+[below](#localisation).
+
+The registration event is the one being signed by [trust
```
Comment (GordonF42) [inline content/information-distribution/source.md]: What do you mean by "registration event" ? the first `network.informo.source` of a source, or any `network.informo.source` event sent by the source?

It's also a bit unclear what the signature generated by the TA will be used for.

File: content/information-distribution/source.md [hunk 12]
```diff
@@ -23,12 +58,221 @@ one key vs all of the keys vs the source's entire Matrix account), and take the
...
+
+## Localisation
+
+An information website might want to publish articles in more than one language.
+In such case, it **can** create sub-sources, each of which handling the
+publication of its articles in one language. A source, regardless of whether it
+is a sub-source or not, **must not** publish articles in more than one language.
+
+A sub-source is an entity similar to a source, with the exception that it
+**must** be related to a source. Each sub-source **must** be a registered Matrix
```
Comment (GordonF42) [inline content/information-distribution/source.md]: Maybe replace _related_ with _linked_ or _tied_ ?

File: content/information-distribution/source.md [hunk 13]
```diff
@@ -4,13 +4,48 @@ weight: 2
...
+through the Matrix room. Each source **must** be a registered Matrix user. A
+Matrix user **must not** register more than one source.
+
+Before publishing any information, a source **must** register itself as such on
+the room. This **must** be done through the publication of a state event of the
+`network.informo.source` type. The event's state key **must** be the ID of the
+source's Matrix user. The content of this event **must** be provided using the
+following model:
+
+|      Parameter      |    Type    | Req. |                                                        Description                                                       |
```
Comment (GordonF42) [inline content/information-distribution/source.md]: This table needs a `network.informo.source` title for better referencing

File: content/information-distribution/source.md [hunk 14]
```diff
@@ -23,12 +58,221 @@ one key vs all of the keys vs the source's entire Matrix account), and take the
...
+A sub-source is an entity similar to a source, with the exception that it
+**must** be related to a source. Each sub-source **must** be a registered Matrix
+user, and **must** register itself on the Matrix room through the publication of
+a state event of the `network.informo.subsource` type, with the Matrix user's ID
+as the event's state key. The event's content **must** be embedded in a [signed
+Matrix event](/information-distribution/signature/#signed-matrix-event), signed
+by one of the parent source's public keys, with its `signed` object using the
+following model:
+
+|      Parameter      |    Type    | Req. |                                  Description                                   |
```
Comment (GordonF42) [inline content/information-distribution/source.md:91]: This table needs a `network.informo.subsource` title for better referencing

File: content/information-distribution/source.md [hunk 15]
```diff
@@ -23,12 +58,221 @@ one key vs all of the keys vs the source's entire Matrix account), and take the
...
+
+An information website might want to publish articles in more than one language.
+In such case, it **can** create sub-sources, each of which handling the
+publication of its articles in one language. A source, regardless of whether it
+is a sub-source or not, **must not** publish articles in more than one language.
+
+A sub-source is an entity similar to a source, with the exception that it
+**must** be related to a source. Each sub-source **must** be a registered Matrix
+user, and **must** register itself on the Matrix room through the publication of
+a state event of the `network.informo.subsource` type, with the Matrix user's ID
```
Comment (GordonF42) [inline content/information-distribution/source.md]: Rewording:
```md
Each sub-source **must** be a registered Matrix user, and **must** register
itself as a sub-source on the Matrix room through the publication of a
`network.informo.subsource` state event, with the Matrix user's ID as the
event's state key
```

File: content/information-distribution/source.md [hunk 16]
```diff
@@ -23,12 +58,221 @@ one key vs all of the keys vs the source's entire Matrix account), and take the
...
+by one of the parent source's public keys, with its `signed` object using the
+following model:
+
+|      Parameter      |    Type    | Req. |                                  Description                                   |
+| ------------------- | ---------- | :--: | ------------------------------------------------------------------------------ |
+| `sig_algo`          | `string`   |  x   | Algorithm the sub-source will use to cryptographically sign its articles. 🔧    |
+| `sig_keys`          | `[string]` |  x   | Public keys the sub-source will use to cryptographically sign its articles. 🔧  |
+| `website`           | `string`   |      | URL of the source's website in this language, if there's one.                  |
+| `description`       | `string`   |      | Short localised description of the source and its publications.                |
+| `custom`            | `object`   |      | Additional information for custom client implementations.                      |
```
Comment (GordonF42) [inline content/information-distribution/source.md:98]: I think we should add a `super_source` field (or something like that). This would make much easier for implementations to retrieve the super source (otherwise they would need to parse the entire source list).

File: content/information-distribution/source.md [hunk 17]
```diff
@@ -4,13 +4,48 @@ weight: 2
...
+the room. This **must** be done through the publication of a state event of the
+`network.informo.source` type. The event's state key **must** be the ID of the
+source's Matrix user. The content of this event **must** be provided using the
+following model:
+
+|      Parameter      |    Type    | Req. |                                                        Description                                                       |
+| ------------------- | ---------- | :--: | ------------------------------------------------------------------------------------------------------------------------ |
+| `name`              | `string`   |  x   | Name of the source.                                                                                                      |
+| `origin`            | `string`   |  x   | The company or individual maintaining this source.                                                                       |
+| `langs`             | `[lang]`   |  x   | Languages of the source's publications.                                                                                  |
```
Comment (GordonF42) [inline content/information-distribution/source.md]: s/langs/localized_sources/ ?

File: content/information-distribution/source.md [hunk 18]
```diff
@@ -4,13 +4,48 @@ weight: 2
...
+| `sig_keys`          | `[string]` |  x   | Public keys the source will use to cryptographically sign its articles. 🔧                                                |
+| `website`           | `string`   |      | URL of the source's website, if there's one.                                                                             |
+| `description`       | `string`   |      | Short description of the source and its publications.                                                                    |
+| `logo`              | `string`   |      | Logo of the source. If provided, must be a [`mxc://` URL](https://matrix.org/docs/spec/client_server/r0.4.0.html#id112). |
+| `custom`            | `object`   |      | Additional information for custom client implementations.                                                                |
+
+ <!-- 🔧: Need to do some research on Megolm and Matrix APIs around encryption
+ and key management -->
+
+Where `lang` is an object using the following model:
```
Comment (GordonF42) [inline content/information-distribution/source.md]: Why not relying on a mapping rather than an array of objects ?

File: content/information-distribution/source.md [hunk 19]
```diff
@@ -23,12 +58,221 @@ one key vs all of the keys vs the source's entire Matrix account), and take the
...
+Once all of these events have been published, `@acmenewsen:example.com` can
+start publishing articles in English, and `@acmenewsfr:example.com` can start
+publishing articles in French.
+
+## Alternate example
+
+Let's consider a news website, named "ACME News", publishing news in both
+English and French, each on a localised website. We'll also consider
+`@acmenews:example.com` its main Informo source, and `@acmenewsfr:example.com`
+its sub-sources, handling the publications of articles in French. The
```
Comment (GordonF42) [inline content/information-distribution/source.md]:
```suggestion
its sub-source, handling the publications of articles in French. The
```

**Review by GordonF42 — CHANGES_REQUESTED** [review]
I propose we add a LocalisedString object, that maps language code to string text, to use for some source fields like name & description, and eventually other fields or other events' fields.

We also need to add a section about source fields that can be overridden by sub-source fields (namely `website` and `description`, and `custom` being implementation dependant)

Also I haven't seen anything about the network.informo.source event being signed by the source user. My first idea was to sign the event using one of the source's previous keys (or any key if it's the first source event), but it requires fetching the entire state event history in order to check if a source event is correctly signed.

File: content/information-distribution/source.md (hunk 1 above)
Comment (vabd) [reply content/information-distribution/source.md]: In this case the limitation *is* the rule imho.

File: content/information-distribution/source.md (hunk 2 above)
Comment (vabd) [reply content/information-distribution/source.md]: Indeed. Though I'd consider this piece of info as belonging in the source's description rather than the article's.

File: content/information-distribution/source.md (hunk 3 above)
Comment (vabd) [reply content/information-distribution/source.md]: Agreed. Will go with `owner`.

File: content/information-distribution/source.md (hunk 4 above)
Comment (vabd) [reply content/information-distribution/source.md]: Good idea.

File: content/information-distribution/source.md (hunk 7 above)
Comment (vabd) [reply content/information-distribution/source.md]: I'm unsure about that, given that the table is instantly followed by

> More information on localisation is available [below](#localisation).

Isn't that enough in your opinion?

File: content/information-distribution/source.md (hunk 8 above)
Comment (vabd) [reply content/information-distribution/source.md]: Good idea.

File: content/information-distribution/source.md (hunk 9 above)
Comment (vabd) [reply content/information-distribution/source.md]: :+1:

File: content/information-distribution/source.md (hunk 10 above)
Comment (vabd) [reply content/information-distribution/source.md]: Agreed. Plus the whole trust management system must be considered optional. I'm rewording this into

> Every trust authority certifying a source's trustworthiness **must** be
> operating by an organism or individual the source trusts and is in contact with
> outside of Informo.

File: content/information-distribution/source.md (hunk 11 above)
Comment (vabd) [reply content/information-distribution/source.md]: 1. Any `network.informo.source` event sent by the source. I'm rewording this so it's more obvious.
2. Ack, I'm adding an explanation to that too. FTR it's for trust management. The link in here refers to information doesn't exist yet (either as a proposal or as part of the specs), which is TAs registration and certification of sources' trustworthiness.

File: content/information-distribution/source.md (hunk 12 above)
Comment (vabd) [reply content/information-distribution/source.md]: Rewording as

> it **must** be referenced to by a source

File: content/information-distribution/source.md (hunk 13 above)
Comment (vabd) [reply content/information-distribution/source.md]: :+1:

File: content/information-distribution/source.md (hunk 14 above)
Comment (vabd) [reply content/information-distribution/source.md:91]: :+1:

File: content/information-distribution/source.md (hunk 15 above)
Comment (vabd) [reply content/information-distribution/source.md]: Gotcha. Doing the same for `network.informo.source`.

File: content/information-distribution/source.md (hunk 16 above)
Comment (vabd) [reply content/information-distribution/source.md:98]: I'd assume that case wouldn't happen very often, since in most cases you'd look for a localised version of a source (so going from the source to the sub-source) rather than for what source does a localised version belong.

I'd picture client implementations only going either "here are all of the localisations of that source" or "here are all of the sources that offer a localisation for that language", both of them being doable by processing only `network.informo.source` events.

IMHO, sub-sources would really only exist as a way to retrieve articles in a specific language rather than anything else.

File: content/information-distribution/source.md (hunk 17 above)
Comment (vabd) [reply content/information-distribution/source.md]: I'd rather go for `l10n`, which is more obvious than `langs` (and I don't really like the naming of `localized_sources` because it refers to sub-sources as sources).

File: content/information-distribution/source.md (hunk 19 above)
Comment (vabd) [reply content/information-distribution/source.md]: Good catch. Apparently I copy-pasted too hastily.

File: content/information-distribution/source.md (hunk 18 above)
Comment (vabd) [reply content/information-distribution/source.md]: I considered both, and went with this one for no real reason, with the aim to have it changed if that point was raised during review. I'm changing it to a `lang:mxid` map.

**Comment by vabd** [comment]
> Also I haven't seen anything about the network.informo.source event being signed by the source user. My first idea was to sign the event using one of the source's previous keys (or any key if it's the first source event), but it requires fetching the entire state event history in order to check if a source event is correctly signed.

From what I remember from our discussions about this, we concluded that the `network.informo.source` event wasn't containing signatures of itself generated by TAs. Instead, each TA must hold a signature of the latest registration event of every source it trusts. Otherwise, it would make having the case of having a source trusted by more that one TA nightmarish. This means that this piece of info belongs to an upcoming SCS focused on TAs.

File: content/information-distribution/source.md (hunk 11 above)
Comment (vabd) [reply content/information-distribution/source.md]: Update: I'm not adding an explanation on how client implementations can use signatures as this piece of info belongs in the TAs section.

File: content/information-distribution/source.md (hunk 1 above)
Comment (GordonF42) [reply content/information-distribution/source.md]: At fist glance I thought you explained the same rule with two different approaches, which is not the case, as "Each source **must** be a registered Matrix user" could be considered true if a user was acting as multiple sources.

I think the underlying rule is something like "sources are special users" or "there is a 1 to 1 link between a user and a source". Tell me what you think ;)

Comment (vabd) [reply content/information-distribution/source.md]: The intention here was actually to explain that 1 to 1 link from each side in order to make sure it couldn't be misunderstood.

**Comment by vabd** [comment]
The SCS should be ready for another review. Please let me know if you think of anything to add under "Client implemetations behaviours regarding sources".

File: content/information-distribution/source.md (hunk 7 above)
Comment (GordonF42) [reply content/information-distribution/source.md]: maybe `**Can** be the current source, more information [below](#localisation).` inside the table cell

or just `*More information [below](#localisation).`

File: content/information-distribution/source.md [hunk 20]
```diff
@@ -4,31 +4,301 @@ weight: 2
...
+signature lives in a TA's registration event 📝.
+
+If a source doesn't provide a logo, client implementations **can** use the
+[avatar](https://matrix.org/docs/spec/client_server/r0.4.0.html#get-matrix-client-r0-profile-userid-avatar-url)
+of its Matrix user in lieu of it if its avatar has been set.
+
+## Compromission of cryptographic private keys
+
+Every trust authority certifying a source's trustworthiness **must** be
+operating by an organism or individual the source trusts and is in contact with
```
Comment (GordonF42) [inline content/information-distribution/source.md]:
```suggestion
operated by an organism or individual the source trusts and is in contact with
```

File: content/information-distribution/source.md (hunk 16 above)
Comment (GordonF42) [reply content/information-distribution/source.md:98]: I see two ways for implementing users' source subscriptions:
1. the client stores a list of matrix IDs containing both supersources and subsources. In this case you'd need some kind of link to go from the subsource to it's supersource to do the json field overriding or to list alternative languages for the subsource.
2. the client stores a list of matrix IDs containing only supersources and a locale key. In this case we wouldn't need a subsource to supersource link.

My approach was to provide a way to go from the subsource to the supersource in case we'd need it later.

I let you decide on this. We can always add it later during the spec process or first client implementation if needed.

File: content/information-distribution/source.md [hunk 21]
```diff
@@ -4,31 +4,301 @@ weight: 2
...
+    "website": "https://www.example.com",
+    "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
+    "logo": "mxc://weu.informo.network/AtEuTuVSeIlZQgjEzjGyMDtG",
+    "sig_algo": "ed25519",
+    "sig_keys": [
+        "IlRMeOPX2e0MurIyfWEucYBRVOEEUMrOHqn/8mLqMjA"
+    ],
+    "lang": {
+        "en": "@acmenews:example.com",
+		"fr": "@acmenewsfr:example.com"
```
Comment (GordonF42) [inline content/information-distribution/source.md]:
```suggestion
        "fr": "@acmenewsfr:example.com"
```

File: content/information-distribution/source.md (hunk 1 above)
Comment (GordonF42) [reply content/information-distribution/source.md]: I'd rephrase it `Each source **must** be a registered Matrix user, and a Matrix user **must not** register more than one source, i.e. there is a one to one correspondence between a source and its Matrix user.` just to be sure

File: content/information-distribution/source.md [hunk 22]
```diff
@@ -4,31 +4,301 @@ weight: 2
...
+through the publication of a `network.informo.source` state event. The event's
+state key **must** be the ID of the source's Matrix user. The content of this
+event **must** be provided using the following model:
+
+## Matrix event `network.informo.source`
+
+|      Parameter      |        Type       | Req. |                                                              Description                                                            |
+| ------------------- | ----------------- | :--: | ----------------------------------------------------------------------------------------------------------------------------------- |
+| `name`              | `localisedString` |  x   | Name of the source.                                                                                                                 |
+| `owner`             | `string`          |  x   | The company or individual maintaining this source.                                                                                  |
```
Comment (GordonF42) [inline content/information-distribution/source.md]: maybe make it a localisedString too? (like if a they want to display their name in their native language version and a english/latin friendly version)

File: content/information-distribution/source.md [hunk 23]
```diff
@@ -4,31 +4,301 @@ weight: 2
...
+* `lang` is a map associating a [RFC
+  5646](https://tools.ietf.org/html/rfc5646)-compliant language (and variant)
+  identifier to the Matrix user ID of the sub-source that handles the
+  publication of articles in this language (and variant). This map **must**
+  contain at least one element. More information on localised sub-sources and
+  examples are available [below](#localisation).
+
+Each time one of the source's properties changes, it **must** publish a new
+registration event, and every trust authority certifying the trustworthiness of
+this source **must** update their signature by basing it on the new event. This
```
Comment (GordonF42) [inline content/information-distribution/source.md]:
```suggestion
this source **must** update their signature to match the new event. This
```

File: content/information-distribution/source.md [hunk 24]
```diff
@@ -4,31 +4,301 @@ weight: 2
...
+signature lives in a TA's registration event 📝.
+
+If a source doesn't provide a logo, client implementations **can** use the
+[avatar](https://matrix.org/docs/spec/client_server/r0.4.0.html#get-matrix-client-r0-profile-userid-avatar-url)
+of its Matrix user in lieu of it if its avatar has been set.
```
Comment (GordonF42) [inline content/information-distribution/source.md]:
```suggestion
of its Matrix user instead.
```

I don't really like the `in lieu of it if its` (hard to read and to say).

Also I don't feel like saying `if its avatar has been set` is really useful, since it's already an optional fallback.

File: content/information-distribution/source.md [hunk 25]
```diff
@@ -4,31 +4,301 @@ weight: 2
...
+authority trusting the source **must** compute and issue a new signature taking
+the updated list of keys into account. The reason behind this is to encourage
+trust authorities to communicate with their trusted sources, estimate how much
+compromised the source is (i.e. one key vs all of the keys vs the source's
+entire Matrix account), and take the actions it deems necessary.
 
 In such an event, client implementations **should** consider articles posted
-prior to the key being compromised as possibly, but not surely trustworthy.
+prior to the key being declared as compromised as possibly, but not surely
+trustworthy.
```
Comment (GordonF42) [inline content/information-distribution/source.md]: I'm not sure of the meaning of this sentence.

I'd phrase it `consider articles posted prior to the key being declared as possibly compromised and not fully trustworthy.`, but I'm probably missing something.

File: content/information-distribution/source.md [hunk 26]
```diff
@@ -4,31 +4,301 @@ weight: 2
...
+| `sig_keys`          | `[string]` |  x   | Public keys the sub-source will use to cryptographically sign its articles. 🔧  |
+| `website`           | `string`   |      | URL of the source's website in this language, if there's one.                  |
+| `description`       | `string`   |      | Short localised description of the source and its publications.                |
+| `custom`            | `object`   |      | Additional information for custom client implementations.                      |
+
+The parent source **must** then reference the sub-source in its own registration
+event, as a `lang` object. The `lang` object **can** reference the source that
+emitted the `network.informo.source` event, but a Matrix user ID **must not** be
+referenced more than once in a source registration's whole array of `lang`
+objects.
```
Comment (GordonF42) [inline content/information-distribution/source.md]: you changed the `lang` array to be a map of language code => mxid ;)

File: content/information-distribution/source.md [hunk 27]
```diff
@@ -4,31 +4,301 @@ weight: 2
...
+
+{{% notice info %}}
+A source **can** register itself as one if its own sub-sources. In this case, it
+doesn't need to emit any `network.informo.subsource` event for this specific
+sub-source. The articles published by the source acting as one of its
+sub-sources **must** then be signed using the one of the source's public keys.
```
Comment (GordonF42) [inline content/information-distribution/source.md]:
```suggestion
sub-sources **must** be signed using one of the source's public keys.
```

File: content/information-distribution/source.md [hunk 28]
```diff
@@ -4,31 +4,301 @@ weight: 2
...
+{{% /notice %}}
+
+If set, client implemetations **must** use the value for the `description` and
```
Comment (GordonF42) [inline content/information-distribution/source.md]:
```suggestion
If set, client implementations **must** use the value for the `description` and
```

File: content/information-distribution/source.md [hunk 29]
```diff
@@ -4,31 +4,301 @@ weight: 2
...
+`website` keys of the `network.informo.subsource` event instead of the localised
+description and the website provided in the parent source's
+`network.informo.source` event.
+
+Client implementations **must** consider a sub-source as holding the same [trust
+level](/trust-management/trust-level) as its parent source, and therefore
+**must** consider it at the same location in trust networks as its parent.
```
Comment (GordonF42) [inline content/information-distribution/source.md:119]:
```suggestion
**must** consider it at the same location in the trust network as its parent.
```

File: content/information-distribution/source.md [hunk 30]
```diff
@@ -4,31 +4,301 @@ weight: 2
...
+
+### Example
+
+Considering the example of an example website publishing news only in English,
```
Comment (GordonF42) [inline content/information-distribution/source.md]:
```suggestion
Consider an example website publishing news only in English,
```

File: content/information-distribution/source.md [hunk 31]
```diff
@@ -4,31 +4,301 @@ weight: 2
...
+```
+
+{{% notice tip %}}
+In this example, `en-US`, `en-GB`, etc. can be used instead of `en` if the
+source wants to explicitely specify language variants.
+{{% /notice %}}
+
+## Client implemetations behaviours regarding sources
+
+Client implementations **must** allow users to subscribe to all of the articles
```
Comment (GordonF42) [inline content/information-distribution/source.md]: I don't feel like this is a relevant obligation for client implementations: clients should be able to provide any kind of subscriptions to a source, like:
- All articles
- Articles published by specific authors
- Articles containing keywords
- ...

IMHO subscribing to all articles from a source is a feature clients should implement in order to cover the standard use case of informo, but they should also be allowed to implement other subscription techniques that could be more restrictive/secure/better than our standard "yolo get all the articles" solution

File: content/information-distribution/source.md [hunk 32]
```diff
@@ -4,31 +4,301 @@ weight: 2
...
+published by a specific source.
+
+Client implemetations **can** display a source and all of its sub-sources as a
+single entity with several languages available.
+
+Client implemetations **should** allow users to set at least one preferred
+language and use it to select the corresponding name and description for sources
+that offer a name or a description in this language. If a source doesn't offer
+either a name or a description in the user's preferred language, client
+implementations **can** select another language to fall back to by allowing the
```
Comment (GordonF42) [inline content/information-distribution/source.md]:
```suggestion
implementations **can** select another language to fall back to, by allowing the
```

File: content/information-distribution/source.md [hunk 33]
```diff
@@ -4,31 +4,301 @@ weight: 2
...
+user to set a weighted list of preferred languages. If a source only provide its
+name or description in one language, client implemetations **must** use that
```
Comment (GordonF42) [inline content/information-distribution/source.md]:
```suggestion
name or description in one language, client implementations **must** use that
```

File: content/information-distribution/source.md [hunk 34]
```diff
@@ -4,31 +4,301 @@ weight: 2
...
+value.
+
+## Full example
+
+Let's consider a news website, named "ACME News", publishing news in both
+English and French, each on a localised website. We'll also consider
+`@acmenews:example.com` its main Informo source, and `@acmenewsen:example.com`
+and `@acmenewsfr:example.com` its sub-sources, handling the publications of
```
Comment (GordonF42) [inline content/information-distribution/source.md]:
```suggestion
and `@acmenewsfr:example.com` its sub-sources, handling the publication of
```

**Review by GordonF42 — CHANGES_REQUESTED** [review]

File: content/information-distribution/source.md (hunk 23 above)
Comment (vabd) [reply content/information-distribution/source.md]: Not sure about the phrasing of a signature matching something (as imho matching means equity, while signatures are either based or generated from content).

File: content/information-distribution/source.md (hunk 26 above)
Comment (vabd) [reply content/information-distribution/source.md]: Whoops, forgot this one.

File: content/information-distribution/source.md (hunk 29 above)
Comment (vabd) [reply content/information-distribution/source.md:119]: Given that we already define a trust network as being either relative to an entity or global to the federation, the original phrasing makes the indication more generic imho.

File: content/information-distribution/source.md (hunk 30 above)
Comment (vabd) [reply content/information-distribution/source.md]: I don't like the idea of addressing the reader directly. I rephrased it as `Let's consider`.

File: content/information-distribution/source.md (hunk 16 above)
Comment (vabd) [reply content/information-distribution/source.md:98]: Gotcha. I'll add it just in case.

File: content/information-distribution/source.md (hunk 1 above)
Comment (vabd) [reply content/information-distribution/source.md]: Sounds good to me.

File: content/information-distribution/source.md (hunk 22 above)
Comment (vabd) [reply content/information-distribution/source.md]: Good idea.

File: content/information-distribution/source.md (hunk 23 above)
Comment (vabd) [reply content/information-distribution/source.md]: FTR I rephrased it as `**must** generate a new signature from the new event`.

File: content/information-distribution/source.md (hunk 25 above)
Comment (vabd) [reply content/information-distribution/source.md]: Let's consider the following example:

* Source `A` publish article `foo`
* Source `A`'s only public key gets compromised and has to change it

Here, `foo` will need to be considered as possibly not having been tampered with, but we can't be sure of that.

File: content/information-distribution/source.md (hunk 31 above)
Comment (vabd) [reply content/information-distribution/source.md]: Right, changing this into a `should`.

**Comment by vabd** [comment]
All of your concerns should have been addressed now.

File: content/information-distribution/source.md (hunk 7 above)
Comment (vabd) [reply content/information-distribution/source.md]: FTR this has already been addressed (imho) when changing the structure of the `lang` object.

File: content/information-distribution/source.md (hunk 30 above)
Comment (GordonF42) [reply content/information-distribution/source.md]: even better :)

**Review by GordonF42 — APPROVED** [review]
Looks good for me :)

**Comment by vabd** [comment]
:tada: :tada: :tada:
